cmd/installer/pkg/install.go

```diff
@@ -29,12 +30,21 @@ func Install(p runtime.Platform, config runtime.Configurator, sequence runtime.S
 	cmdline := kernel.NewCmdline("")
 	cmdline.Append("initrd", filepath.Join("/", "default", constants.InitramfsAsset))
 	cmdline.Append(constants.KernelParamPlatform, p.Name())
-	cmdline.Append(constants.KernelParamConfig, opts.ConfigSource)
 
 	if err = cmdline.AppendAll(p.KernelArgs().Strings()); err != nil {
 		return err
 	}
 
+	option := cmdline.Get(constants.KernelParamConfig).First()
+
+	switch {
+	case option == nil:
+		cmdline.Append(constants.KernelParamConfig, opts.ConfigSource)
+	case option != nil && opts.ConfigSource != "none":
```

Comment:
Not sure I understand this. The `option` variable is not `nil`, yet the warning says "no kernel arguments supplied".